### PR TITLE
Update to OpenAPI generator v2

### PR DESCRIPTION
**Pull request by theSoenke**
- Upgrade to the generator package v2 (but keep the generator cli at 4.3.1 to avoid breaking changes)
- Add missing resources to schema and fix linting warnings
- Update licenses

### CODE REVIEW

File: schemas.yaml [hunk 1]
```diff
@@ -110,6 +110,14 @@ schemas:
     "$ref": schemas/job_preview.yaml#/job_preview
   job_locale:
     "$ref": schemas/job_locale.yaml#/job_locale
+  job_template:
```
Comment (jablan) [inline schemas.yaml:113]: how come we were missing all these?

File: openapi-generator/go_lang.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 generatorName: go
 outputDir: clients/go
 packageName: phrase
-packageVersion: 2.6.5
+packageVersion: 2.6.6
```
Comment (jablan) [inline openapi-generator/go_lang.yaml:5]: is the version bump just due to tooling upgrade? should there be any actual change in the generated library?

**Review by jablan — APPROVED** [review]

File: openapi-generator/go_lang.yaml (hunk 2 above)
Comment (jablan) [reply openapi-generator/go_lang.yaml:5]: ah I see the comment now.

File: schemas.yaml (hunk 1 above)
Comment (theSoenke) [reply schemas.yaml:113]: honestly i don't know either. running `npx openapi-generator-cli validate -i tmp/compiled.yaml` will find all of these issues but it also finds a few more deprecated things i've not been able to fix so far

**Pull request merged**